scopes/workspace/workspace/workspace.ts

```diff
@@ -462,11 +463,11 @@ export class Workspace implements ComponentFactory {
     const results: Array<{ extensionId: string; results: SerializableResults }> = [];
     await mapSeries(onChangeEntries, async ([extension, onChangeFunc]) => {
       const onChangeResult = await onChangeFunc(component);
-      // TODO: find way to standardize event names.
-      await this.graphql.pubsub.publish(ComponentChanged, { componentChanged: { component } });
       results.push({ extensionId: extension, results: onChangeResult });
     });
 
+    // TODO: find way to standardize event names.
+    await this.graphql.pubsub.publish(ComponentChanged, { componentChanged: { component } });
```

Comment:
this published the `ComponentChanged` event many times in the loop, moved it outside to publish only once.
@guysaar223 - what do you think?